kpatch-build/kpatch-elf.c

```diff
@@ -759,7 +759,8 @@ void kpatch_rebuild_rela_section_data(struct section *sec)
 		ERROR("size mismatch in rebuilt rela section");
 }
 
-void kpatch_write_output_elf(struct kpatch_elf *kelf, Elf *elf, char *outfile)
+void kpatch_write_output_elf(struct kpatch_elf *kelf, Elf *elf, char *outfile,
+			     int flags)
```

Comment:
So, this is going to be nitpicking a bit. But the `flags` argument will be only used for the `mode` argument of `creat`.
This might get confusing when you see that `open` (of which creat is just an specialised alias) also has a `flags` argument whose value don't match the `mode` argument.

So, could this be called `file_mode` or something mode indicative?

Reply:
Sure, let's rename it to be clear.  And good catch on flags vs mode, I copied the wrong name from the manpage.